docker/postgres.dockerfile

```diff
@@ -0,0 +1,34 @@
+ARG BASE_IMAGE_NAME=ghcr.io/batteries-included/build-base
+ARG BASE_IMAGE_TAG=latest
+
+ARG DEPLOY_IMAGE_NAME=ghcr.io/batteries-included/deploy-base
+ARG DEPLOY_IMAGE_TAG=latest
+
+
+FROM ${DEPLOY_IMAGE_NAME}:${DEPLOY_IMAGE_TAG}
+
+ARG PG_VERSION=latest
+ARG PG_MAJOR
+
+RUN apt-get update && \
+    apt-get install -y \
+        --no-install-recommends \
+        gnupg \
+        postgresql-common && \
+    /usr/share/postgresql-common/pgdg/apt.postgresql.org.sh -y -c "${PG_MAJOR}" && \
+    apt-get install -y \
+        --no-install-recommends \
+        -o Dpkg::::="--force-confdef" \
+        -o Dpkg::::="--force-confold" \
+        postgresql-common && \
+    sed -ri 's/#(create_main_cluster) .*$/\1 = false/' /etc/postgresql-common/createcluster.conf && \
+    apt-get install -y \
+        --no-install-recommends \
+        -o Dpkg::::="--force-confdef" \
+        -o Dpkg::::="--force-confold" \
+        "postgresql-${PG_MAJOR}=${PG_VERSION}*" && \
```

Comment:
While we are here, we should install:

- pgaudit
- pgvector
- pg-failover-slots
- pg_stat_monitor
- postgis-3

They seem to be in the form of `postgres-${PG_MAJOR}-extension-name`

Reply:
pg_stat_monitor isn't available in the standard packages and isn't installed in the cnpg images. We can add that later but I'd like to get it working initially and land it before jumping into additional features. Everything else is added now.

Reply:
Totally, I was just going by this: https://www.postgresql.org/download/products/6-postgresql-extensions/